src/Model/Filter/Base.php

```diff
@@ -208,12 +209,55 @@ protected function passedValue(): mixed
         }
 
         if ($this->getConfig('multiValueSeparator')) {
-            return explode($this->getConfig('multiValueSeparator'), $value);
+            if (!$this->getConfig('multiValueExactMatching')) {
+                return explode($this->getConfig('multiValueSeparator'), $value);
+            }
+
+            return $this->parseSearchTerms($value);
         }
 
         return $value;
     }
 
+    /**
+     * @param string $input
+     * @return array<string>
+     */
+    protected function parseSearchTerms(string $input): array
```

Comment:
Why is all this parsing code in `Base` when `multiValueSeparator` and `multiValueExactMatching` options are only relevant for `Value` filter?

Reply:
I guess I can look into any potential refactoring after merging this.